src/services/jsTyping.ts

```diff
@@ -46,8 +55,44 @@ namespace ts.JsTyping {
             }
         }
         
-        let newTypingNames = typingNames.filter(t => cachedTypingNames.indexOf(t) < 0);
-        
+        let newTypingNames = ts.filter(typingNames, t => cachedTypingNames.indexOf(t) < 0);
         return { cachedTypingPaths, newTypings: newTypingNames };
     }
+
+    export function updateTypingsConfig(
+        host: TypeDefinitionResolutionHost | System, cachedTypingsPaths: string[], newTypings: string[], cachePath: string, typingsConfigPath: string): void {
+        let tsdJsonCacheInfo = tryGetTsdJsonCacheInfo(host, cachePath);
+        if (tsdJsonCacheInfo) {
+            let notFound: string[] = [];
+            let cacheTsdJsonDict = tsdJsonCacheInfo.value;
+            if (cacheTsdJsonDict.hasOwnProperty("installed")) {
+                let installedTypings = Object.keys(cacheTsdJsonDict.installed);
+                notFound = ts.filter(newTypings, i => !isInstalled(i, installedTypings));
+            }
+            if (cacheTsdJsonDict.hasOwnProperty("notFound")) {
+                notFound = ts.deduplicate<string>(cacheTsdJsonDict["notFound"].concat(notFound));

```

Comment:
I think you're right - we should be able to simplify to: notFound = cacheTsdJsonDict["notFound"].concat(notFound) 
At the time I was testing it I didn't have the "notFound" filtering in discoverTypings so you were able to get duplicates.  This shouldn't happen now that the filter is in place. The plan is to clear the notFound field as part of the updateTypings so we should be good on that front.